feat: session segment updates

nwykes opened the pull request:
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### Description

only display UserInfoSeparator if both username and hostname are dislayed
cleanHostName only if it ends with garbage instead of contains.  fixes .localdomain having just .local stripped
add .localdomain to garbage

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

JanJoris commented:
Nobody accused anyone of being dumb here. If it was perceived that way, I should apologise. Not you. I need to learn to stick by my own guidelines. If one user reports an issue, chances are he is not alone in experiencing that specific issue. So, thank you for doing this!

JanJoris commented:
I merged it myself as it needed a bit of rebasing and whatnot. Youc an see it [here](https://github.com/JanJoris/oh-my-posh/commit/b7a46bafe75e6113da9993e781ec6134c32a4e5f)

nwykes committed:
feat: session segment updates

only display UserInfoSeparator if both username and hostname are dislayed
cleanHostName only if it ends with garbage instead of contains.
  fixes .localdomain having just .local stripped
add .localdomain to garbage

JanDeDobbeleer commented on segment_session_test.go:
```diff
@@ -7,7 +7,7 @@ import (
 )
 
-func testUserInfoWriter(userInfoSeparator string, username string, hostname string, goos string) string {
+func setupSession(userInfoSeparator string, username string, hostname string, goos string) session {
```
good one
